src/test/TUSDCurveCallee.t.sol

```diff
@@ -0,0 +1,332 @@
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertEq(Token(dai).balanceOf(address(this)), 0);
+        assertEq(Token(tusd).balanceOf(address(this)), 0);
+    }
+
+    function test_profit() public {
+        uint256 minProfit = 1000 * WAD;
+        uint256 amt = 1_000_000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(minProfit)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertGe(Token(dai).balanceOf(address(123)), minProfit);
+    }
+
+    function test_profitBigAmt() public {
+        uint256 minProfit = 1000 * WAD;
+        uint256 amt = 10_000_000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(minProfit)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertGe(Token(dai).balanceOf(address(123)), minProfit);
+    }
+
+    function test_profitAllCollateral() public {
+        uint256 minProfit = 1000 * WAD;
+        uint256 amt = 26_000_000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(minProfit)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+        address dai = Chainlog(chainlog).getAddress("MCD_DAI");
+        assertGe(Token(dai).balanceOf(address(123)), minProfit);
+    }
+
+    function test_maxPrice() public {
+        uint256 amt = 20000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(this),
+            address(gemJoin),
+            uint256(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        address osm = Chainlog(chainlog).getAddress("PIP_TUSD");
+        Hevm(hevm).store({
```

Comment:
This can go in`setUp()`

Reply:
Moved the osm address retreiving to setup as all the chainlog queries.
Left the taking of ownership here as it is only used in this test.